### PR TITLE
Bare-bones of the app

### PR DESCRIPTION
**Pull request by steve1998**
## Description

Added the base files for the Cuaca app. This includes the HTML, CSS, and JavaScript files for the base app. 

## Related Issue

No related issues.

## Motivation and Context

This is the bare-bones of the app.

## How Has This Been Tested?

No testing has been done yet.

## Screenshots (if appropriate):
cuaca

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: js/weather.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    // create texts and appends it to the message element created
+    var node = document.createTextNode(x + " ");
+    var space = document.createTextNode(" ");
+    var sign = document.createTextNode("C");
+    // var icon = document.createElement("i");
+    var br = document.createElement("br");
+    var img = document.createElement("img");
+    var degree = document.createTextNode("\xB0");
+    degree.src = "https://png.icons8.com/small/50/000000/0-degrees.png";
```
Comment (ferdinh) [inline js/weather.js]: Text node does not have an attribute of `src`. If the goal here is to use the degree symbol provided by icons8 and fallback to unicode degree, then, use `img`.

File: js/weather.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    var elementToAdd = document.createElement("div");
+    elementToAdd.className = "card";
+    // elementToAdd.classList.add("is-primary");
+    // elementToAdd.classList.add("is-large")
+
+    // creates message element in article
+    var child = document.createElement("div");
+    child.className = "card-content";
+
+    // p tag
```
Comment (ferdinh) [inline js/weather.js]: Redundant comment.

File: js/weather.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    newDiv.appendChild(farenheit);
+    
+    // appends article to container
+    document.getElementById("contain").appendChild(elementToAdd);
+    document.getElementById("contain").appendChild(newDiv);
+
+
+} */
+
+function insertWeatherData (x, weather) {
```
Comment (ferdinh) [inline js/weather.js]: I suppose `x` is the temperature. Please rename it for clarity.

Suggestion for `weather`: 

- `condition`
- `weatherCondition`

File: js/weather.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    content.appendChild(sign); // celsius sign
+    content.appendChild(br); // adds break
+    content.appendChild(img); // appends logo
+    content.appendChild(space); // adds blank space
+    content.appendChild(weatherText); // weather info
+
+    // appends divs and set id
+    child.appendChild(content);
+    elementToAdd.appendChild(child);
+    elementToAdd.id = "weatherObject";
```
Comment (ferdinh) [inline js/weather.js]: This can be moved to line 51, after the assignment to `.className`.

File: js/weather.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+    elementToAdd.appendChild(newDiv);    
+    */
+
+    // appends article to container
+    document.getElementById("contain").appendChild(elementToAdd);
+
+}
+
+function convert (x, text){
```
Comment (ferdinh) [inline js/weather.js]: Ambiguous naming.

Suggestion: `convertKelvinTo(temp, degree)`

File: js/weather.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    var final = x - 273.15;
+    if (text == "F") {
+        final = final * (9/5);
```
Comment (ferdinh) [inline js/weather.js]: Use shorthand `*=`.

File: js/weather.js [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        final += 32;
+    }
+    return final;
+}
+
+function removeWeatherData (x) {
```
Comment (ferdinh) [inline js/weather.js]: It is better to rename `x` to `id` or `elementId`.

File: js/weather.js [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+        if(document.getElementById("weatherObject")){
+            var id1 = "weatherObject";
+            var id2 = "celsius";
+            var id3 = "fahrenheit";
+
+            removeWeatherData(id1);
+            removeWeatherData(id2);
+            removeWeatherData(id3);
+        }
```
Comment (ferdinh) [inline js/weather.js]: You can use direct string declaration. e.g `removeWeatherData("weatherObject")`.

Method call for `id2` and `id3` seems redundant as I found no element with said `id`?

File: js/weather.js [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+
+        insertWeatherData(temp, weather);
+    })
+}
+
+var input = document.getElementById("input");
+
+input.addEventListener("keydown", function (a) {
+    if (a.keyCode === 13) {  // checks whether the pressed key is "Enter"
```
Comment (ferdinh) [inline js/weather.js]: 'When "enter" is pressed' will be enough for the comment.

File: index.html [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+              <p> Powered by <a href="https://openweathermap.org"><strong>OpenWeather</strong></a></p>
+            </div>
+            <div class="content has-text-centered">
+                <a href="https://icons8.com">Icon pack by <strong>Icons8</strong></a>
+            </div>
+            <div class="content has-text-centered">
+                <i class="fab fa-github" href="https://github.com/wakatobi/cuaca"></i>
+            </div>
+        </footer>
+        <script src="js/weather.js" async defer></script>
```
Comment (ferdinh) [inline index.html]: I think `async` and `defer` here will not make sense unless the `script` tag is placed on the `head`.

File: js/weather.js [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+    var newDiv = document.createElement('div');
+    newDiv.appendChild(celsius);
+    newDiv.appendChild(farenheit);
+    
+    // appends article to container
+    document.getElementById("contain").appendChild(elementToAdd);
+    document.getElementById("contain").appendChild(newDiv);
+
+
+} */
```
[the remark this answers is not in the document]
Comment (steve1998) [reply js/weather.js]: Done

[the remark this answers is not in the document]
Comment (steve1998) [reply js/weather.js]: I've removed the commented code.

File: js/weather.js [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+
+function insertWeatherData (x, weather) {
+    // creates article element to display weather data
+    var elementToAdd = document.createElement("div");
```
[the remark this answers is not in the document]
Comment (steve1998) [reply js/weather.js]: I have changed the variable name and cleared up any mismatches in the comments.

**Comment by ferdinh** [comment]
Merged from PR #5.

**Pull request closed without merging**